lib/improver/nbhood/use_nbhood.py

```diff
@@ -156,3 +159,133 @@ def process(self, cube, mask_cube):
+                for masking.
+
+        Keyword Args:
+            weights (cube):
+                A cube from an ancillary file containing the weights for each
+                point in the coord_masked at each grid point. Only two points
+                in coord_masked can have a non-zero weight for each grid-point,
+                i.e. we are only weighting between two adjacent bands in the
+                neighbourhood output for each gridpoint.
+                Should have the coordinates coord_masked, x and y.
+                Default is None which equal weights for each band in the mean
+                used to collapse the chosen coordinate.
+                The weights cube can be masked, and this mask will be retained,
+                and will be present in the output.
+.
+        """
+        self.coord_masked = coord_masked
+        self.weights = weights
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<ApplyNeighbourhoodProcessingWithAMask: '
+                  'coord_masked: {}, weights: {}>')
+        return result.format(self.coord_masked,
+                             self.weights)
+
+    def reweight_weights(self, nbhood_cube):
+        """
+        Reweight the weights taking into account where there are NaNs in the
+        result from neighbourhood.
+
+        The weights corresponding to NaNs in the result from neighbourhooding
+        with a mask are set to zero and then the weights are renormalized along
+        the axis corresponding to the coordinate we want to collapse.
+
+        Args:
+            nbhood_cube (iris.cube.Cube):
+                The cube that has been through masked neighbourhood processing
+                and has the dimension we wish to collapse. Must have the same
+                dimensions of the cube.
+        """
+        # If the weights are masked we want to retain the mask.
+        condition = np.isnan(nbhood_cube.data)
+        if ma.is_masked(self.weights.data):
+            condition = condition & ~self.weights.data.mask
+
+        self.weights.data[condition] = 0.0
+        axis = nbhood_cube.coord_dims(self.coord_masked)
+        self.weights.data = WeightsUtilities.normalise_weights(
+            self.weights.data, axis=axis)
+
+    def process(self, cube):
+        """
+        Collapse the chosen coordinates with the available weights. The result
+        of the neighbourhood processing is taken into account to reweight any
+        weights corresponding to a NaN in the result from neighbourhooding. In
+        this case the weights are re-normalized so that we do not lose
+        probability.
+
+        Args:
+            cube (Iris.cube.Cube):
+                Cube containing the array to which the square neighbourhood
+                with a mask has been applied.
+
+        Returns:
+            result (Iris.cube.Cube):
+                Cube containing the weighted result from neighbourhood after
```

Comment:
Could you change this to "weighted mean".

Reply:
Corrected.